cmd/juju/application/store/charmadapter.go

```diff
@@ -48,26 +46,19 @@ type CharmAdaptor struct {
 }
 
 // NewCharmAdaptor returns a CharmAdaptor.
-func NewCharmAdaptor(charmsAPI CharmsAPI, charmStoreRepoFunc CharmStoreRepoFunc, downloadBundleClientFunc DownloadBundleClientFunc) *CharmAdaptor {
+func NewCharmAdaptor(charmsAPI CharmsAPI, downloadBundleClientFunc DownloadBundleClientFunc) *CharmAdaptor {
 	return &CharmAdaptor{
-		charmsAPI:          charmsAPI,
-		charmStoreRepoFunc: charmStoreRepoFunc,
```

Comment:
Remove charmStoreRepoFunc from CharmAdapter, ln 46. And definition of charmStoreRepoFunc at ln 22.